hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java

```diff
@@ -116,6 +115,20 @@ public static HederaEvmTransactionResult failureFrom(final long gasUsed, @NonNul
                 Collections.emptyList());
     }
 
+    public static HederaEvmTransactionResult resourceExhaustionFrom(
+            final long gasUsed, final long gasPrice, @NonNull final ResponseCodeEnum reason) {
+        return new HederaEvmTransactionResult(
+                gasUsed,
+                gasPrice,
+                null,
+                null,
+                Bytes.EMPTY,
+                null,
+                null,
```

Comment:
Is there an appropriate ResponseCode?

Reply:
By "abort" I mean a transaction failing before it even makes it to the stage of initiating a top-level `MESSAGE_CALL` or `CONTRACT_CREATION`.

So in these cases, there is no EVM transaction started; and hence no opportunity to revert or halt.